core/src/test/java/ProcessRunner/ProcessIntegrationTest.java

```diff
@@ -0,0 +1,232 @@
+    }
+
+    @Test
+    void cmdWriteTest() {
+        // 1. Create cmd process
+        ProcessBuilder pb = new ProcessBuilder("cmd");
+        ProcessRunner process = ProcessFactory.createCustomProcess(pb);
+
+        InputReceiver receiver = new InputReceiver();
+        process.addInputObserver(receiver);
+
+        process.start();
+
+        // 2. Wait for process to start
+        try {
+            Thread.sleep(500);
+        } catch (InterruptedException ignored) {
+        }
+
+
+        // 3. Skip beginning lines from cmd
+        receiver.discardBufferedLinesReceived();
+
+
+        // 4. Send "test"
+        try {
+            process.outputToScript("test");
+        } catch (IOException ignored) {
+        }
+
+        // 5. Wait for command to process
+        try {
+            Thread.sleep(500);
+        } catch (InterruptedException ignored) {
+        }
+
+        // 6. Read the first word of the command output
+        String lastReceived;
+        StringBuilder sb = new StringBuilder();
+        while ((lastReceived = receiver.getNextInput()) != null) {
+            if (!lastReceived.equals("\n")) {
+                sb.append(lastReceived);
+            } else {
+                break;
+            }
+        }
+
+        // The first word read should be "test"
+        assert sb.toString().equals("test");
+    }
+
+    @Test
+    void pythonReadTest() throws IOException, InterruptedException {
+        // 1. Create temporary Python file
+        File pyScript = new File("pythonTemp.py");
+        BufferedWriter bf;
+        bf = new BufferedWriter(new FileWriter(pyScript));
+
+        String[] pythonCode = {
+                "import time\n" +
+                        "\n",
+                "while True:\n",
+                "\tprint(\"test\")\n",
+                "\ttime.sleep(0.05)\n",
+        };
+
+        bf.write("");
+        for (String s : pythonCode) {
+            bf.append(s);
+            bf.flush();
+        }
+
+        // 2. Create Python process
+        ProcessRunner process = ProcessFactory.createPythonProcess(pyScript.getAbsolutePath());
+
+        InputReceiver receiver = new InputReceiver();
+        process.addInputObserver(receiver);
+        process.start();
+
+        // 3. Wait for process to start
+        Thread.sleep(500);
+
+        // 3. Read the first word of the command output
+        String lastReceived;
+        StringBuilder sb = new StringBuilder();
+        while (
+                (lastReceived = receiver.getNextInput()) != null &&
+                        !lastReceived.equals(" ") &&
+                        !lastReceived.equals("\n")
+                ) {
+            sb.append(lastReceived);
+        }
+
+        // First word should be "test"
+        assert sb.toString().equals("test");
+    }
+
+    @Test
+    void pythonWriteTest() throws IOException, InterruptedException {
+        // 1. Create temporary Python file
+        File pyScript = new File("pythonTemp.py");
```

Comment:
Rename variable to bw or something that can be associated to the Buffered Writer class? bf seems qute unclear :)